experiments/TestCase/solid_body_rotation_mountain.jl

```diff
@@ -0,0 +1,320 @@
+#!/usr/bin/env julia --project
+using ClimateMachine
+ClimateMachine.init(parse_clargs = true)
+
+using ClimateMachine.Atmos
+using ClimateMachine.Orientations
+using ClimateMachine.ConfigTypes
+using ClimateMachine.NumericalFluxes
+using ClimateMachine.Diagnostics
+using ClimateMachine.GenericCallbacks
+using ClimateMachine.ODESolvers
+using ClimateMachine.TurbulenceClosures
+using ClimateMachine.SystemSolvers: ManyColumnLU
+using ClimateMachine.Mesh.Filters
+using ClimateMachine.Mesh.Grids
+using ClimateMachine.Mesh.Interpolation
+using ClimateMachine.TemperatureProfiles
+using ClimateMachine.VariableTemplates
+using ClimateMachine.Thermodynamics: air_density, total_energy
+
+using LinearAlgebra
+using StaticArrays
+using Test
+
+using CLIMAParameters
+using CLIMAParameters.Planet: day, planet_radius
+struct EarthParameterSet <: AbstractEarthParameterSet end
+const param_set = EarthParameterSet()
+
+function setmax(f, r_inner, r_outer)
+    function setmaxima(a, b, c)
+        return f(
+            a,
+            b,
+            c,
+            max(abs(a), abs(b), abs(c));
+            r_inner = r_inner,
+            r_outer = r_outer,
+        )
+    end
+    return setmaxima
+end
+
+function cubedshelltopowarp(
+    a,
+    b,
+    c,
+    R = max(abs(a), abs(b), abs(c));
+    r_inner = _planet_radius,
+    r_outer = _planet_radius + domain_height,
+)
+
+    function f(sR, ξ, η, faceid)
+        R_m = π * 3 / 4
+        h0 = 2000
+        ζ_m = π / 16
+        φ_m = 0
+        λ_m = π * 3 / 2
+
+        X, Y = tan(π * ξ / 4), tan(π * η / 4)
+
+        # Linear Decay Profile
+        Δ = (r_outer - abs(sR)) / (r_outer - r_inner)
+        δ = 1 + X^2 + Y^2
+
+        # Angles
+        # mR == modified radius 
+        mR = sR
+        if faceid == 1
+            λ = atan(X)                     # longitude 
+            φ = atan(cos(λ) * Y)              # latitude
+        elseif faceid == 2
+            λ = atan(X) + π / 2
+            φ = atan(Y * cos(atan(X)))
+        elseif faceid == 3
+            λ = atan(X) + π
+            φ = atan(Y * cos(atan(X)))
+        elseif faceid == 4
+            λ = atan(X) + (3 / 2) * π
+            φ = atan(Y * cos(atan(X)))
+        elseif faceid == 5
+            λ = atan(X, -Y) + π
+            φ = atan(1 / sqrt(δ - 1))
+        elseif faceid == 6
+            λ = atan(X, Y)
+            φ = -atan(1 / sqrt(δ - 1))
+        end
+
+        r_m = acos(sin(φ_m) * sin(φ) + cos(φ_m) * cos(φ) * cos(λ - λ_m))
```

Comment:
@jiahe23 We could update this to allow a user to simply specify the coordinate dependent radial warps; the face-identification can happen in the inner meshwarp function, in general I imagine we don't need access to that component at the experiment/driver level (?) [i.e. collapse lines 89-106 in some sensible way]